helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -231,8 +231,11 @@ public void dropInstance(String clusterName, InstanceConfig instanceConfig) {
     // delete config path
```

Comment:
nit, these duplicates with purgeInstance(), can we merge them?

I know it will cause two "String instancePath = PropertyPathBuilder.instance(clusterName, instanceName);" in 2 methods. But IMO, it cleans up the code further.

No strong preference here.

Reply:
I'd rather not since it's not a strict duplication: `dropInstance` is based on `InstanceConfig`, `purgeInstance` is based on instance name. We can make `dropInstance` call `purgeInstance` but that's a lot of twist and turns.